src/video/x11/SDL_x11vulkan.c

```diff
@@ -232,4 +232,59 @@ void X11_Vulkan_DestroySurface(SDL_VideoDevice *_this,
     }
 }
 
+SDL_bool X11_Vulkan_GetPresentationSupport(SDL_VideoDevice *_this,
+                                           VkInstance instance,
+                                           VkPhysicalDevice physicalDevice,
+                                           Uint32 queueFamilyIndex)
+{
+    SDL_VideoData *videoData = _this->driverdata;
+    PFN_vkGetInstanceProcAddr vkGetInstanceProcAddr;
+    const char *forced_visual_id;
+    VisualID visualid;
+
+    if (!_this->vulkan_config.loader_handle) {
+        return SDL_SetError("Vulkan is not loaded");
+    }
+    vkGetInstanceProcAddr = (PFN_vkGetInstanceProcAddr)_this->vulkan_config.vkGetInstanceProcAddr;
+
+    forced_visual_id = SDL_GetHint(SDL_HINT_VIDEO_X11_WINDOW_VISUALID);
+    if (forced_visual_id) {
+        visualid = SDL_strtol(forced_visual_id, NULL, 0);
+    } else {
+        visualid = 0;
```

Comment:
I don't know that this is correct, you might need to open the display and get the default visual. See get_visualinfo() and X11_GLES_GetVisual() for the gory details.

Although allowing a visualid of 0 would be an awesome sane default, if it works.

Reply:
Can you test it with visualid 0? Does that actually work?

Reply:
Found some terrifying X macros that allow getting default screens and visuals, pushed something that hypothetically should work...?

Reply:
Added the check to testvulkan and can confim the latest works on Xwayland at least!